ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/metrics_correlation/MetricsCorrelation.java

```diff
@@ -0,0 +1,311 @@
+import org.opensearch.ml.common.MLModel;
+import org.opensearch.ml.common.MLTask;
+import org.opensearch.ml.common.exception.ExecuteException;
+import org.opensearch.ml.common.exception.MLException;
+import org.opensearch.ml.common.input.Input;
+import org.opensearch.ml.common.input.execute.metricscorrelation.MetricsCorrelationInput;
+import org.opensearch.ml.common.model.MLModelConfig;
+import org.opensearch.ml.common.model.MLModelFormat;
+import org.opensearch.ml.common.model.MLModelState;
+import org.opensearch.ml.common.model.MetricsCorrelationModelConfig;
+import org.opensearch.ml.common.output.execute.metrics_correlation.MCorrModelTensors;
+import org.opensearch.ml.common.output.execute.metrics_correlation.MetricsCorrelationOutput;
+import org.opensearch.ml.common.output.model.ModelResultFilter;
+import org.opensearch.ml.common.transport.deploy.MLDeployModelAction;
+import org.opensearch.ml.common.transport.deploy.MLDeployModelRequest;
+import org.opensearch.ml.common.transport.deploy.MLDeployModelResponse;
+import org.opensearch.ml.common.transport.model.MLModelGetAction;
+import org.opensearch.ml.common.transport.model.MLModelGetRequest;
+import org.opensearch.ml.common.transport.model.MLModelGetResponse;
+import org.opensearch.ml.common.transport.model.MLModelSearchAction;
+import org.opensearch.ml.common.transport.register.MLRegisterModelAction;
+import org.opensearch.ml.common.transport.register.MLRegisterModelInput;
+import org.opensearch.ml.common.transport.register.MLRegisterModelRequest;
+import org.opensearch.ml.common.transport.register.MLRegisterModelResponse;
+import org.opensearch.ml.common.transport.task.MLTaskGetAction;
+import org.opensearch.ml.common.transport.task.MLTaskGetRequest;
+import org.opensearch.ml.common.transport.task.MLTaskGetResponse;
+import org.opensearch.ml.engine.algorithms.DLModelExecute;
+import org.opensearch.ml.engine.annotation.Function;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+
+import static org.opensearch.index.query.QueryBuilders.termQuery;
+
+@Log4j2
+@Function(FunctionName.METRICS_CORRELATION)
+public class MetricsCorrelation extends DLModelExecute {
+
+    private static final int AWAIT_BUSY_THRESHOLD = 1000;
+    private Client client;
+    private final Settings settings;
+    //As metrics correlation is an experimental feature we are marking the version as 1.0.0b1
+    public static final String MCORR_ML_VERSION = "1.0.0b1";
+    //This is python based model which is developed in house.
+    public static final String MODEL_TYPE = "in-house";
+    public static final String MCORR_MODEL_URL =
+            "https://artifacts.opensearch.org/models/ml-models/amazon/metrics_correlation/1.0.0b1/torch_script/metrics_correlation-1.0.0b1-torch_script.zip";
```

Comment:
This hard coded URL seems not easy to support new version. For example, if user deploy this code with OS 2.7, then later we released a new version of metrics correlation, user has no way to use the newer version, right?